test/deploy/rackhd_microservice_install.py

```diff
@@ -0,0 +1,185 @@
+
+            # Checking the users with whoami command
+            command = "sudo usermod -aG docker $(whoami)"
+            self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "whoami command failed.")
+
+        # Check if docker-compose is already installed
+        rsp = fit_common.remote_shell("docker-compose -v")
+        if rsp['exitcode'] == 0:
+            logs.info(" docker-compose already installed")
+            if "docker-compose version" in rsp['stdout']:
+                logs.info(" Docker installed: %s", rsp['stdout'])
+        else:
+            # Installing python pip
+            command = PROXYVARS + "sudo apt-get -y install python-pip"
+            self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "Pip install failed.")
+
+            # Installing docker compose
+            logs.info(" Install docker-compose")
+            command = PROXYVARS + "sudo pip install docker-compose"
+            self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "Docker compose install failed.")
+
+        # Setup docker proxy environment
+        logs.info(" Setup Docker Proxy")
+        dockerproxy = open("dockerproxy", 'w')
+        dockerproxy.write('export http_proxy="http://web.hwimo.lab.emc.com:3128/"\n')
+        dockerproxy.close()
+        fit_common.scp_file_to_ora("dockerproxy")
+        self.assertEqual(fit_common.remote_shell('cat dockerproxy >> /etc/default/docker'
+                                                 )['exitcode'], 0, "adding docker proxy config failed.")
+        os.remove('dockerproxy')
+
+        # Restart the docker service
+        command = PROXYVARS + "sudo service docker restart"
+        self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "Docker service restart failed.")
+        time.sleep(60)
+
+    def test04_setup_rackhd_docker_services(self):
+        # add the .env variables for HOST IP into the ".env" file
+        envfile = open("envfile", 'w')
+        envfile.write("TAG=latest\n")
+        envfile.write("REGISTRY_IP=172.31.128.1\n")
+        host_ip = "HOST_IP=" + socket.gethostbyname(fit_common.fitcfg()['rackhd_host']) + "\n"
+        envfile.write(host_ip)
+        envfile.close()
+        fit_common.scp_file_to_ora("envfile")
+        self.assertEqual(fit_common.remote_shell('cp envfile /home/onrack/.env'
+                                                 )['exitcode'], 0, "copy of env file failed.")
+        os.remove('envfile')
+
+        # Get the username and password from config-mn/credentials.json
+        username = fit_common.fitcreds()['docker_hub'][0]['username']
+        password = fit_common.fitcreds()['docker_hub'][0]['password']
+        command = 'cd rackhd/docker/dell; sudo docker login --username=' + username + ' --password=' + password
+        self.assertEqual(fit_common.remote_shell(command)['exitcode'], 0, "Docker login failed.")
+        time.sleep(30)
```

Comment:
Evaluate all of the sleeps in this script.  If the command is synchronous, it shouldn't need a sleep.